File: app/src/main/java/com/sp/base/util/SerializedNameKeys.java
package com.sp.base.util;

public class SerializedNameKeys {
    public static final String key = "key";
    public static final String statusCode = "code";
    public static final String statusMessage = "message";
    public static final String data = "data";
    public static final String id = "id";
    public static final String riderId = "rider_id";
    public static final String latitude = "latitude";
    public static final String longitude = "longitude";
    public static final String address = "address";
    public static final String created = "created_at";
    public static final String updated = "updated_at";
    public static final String locations = "locations";
    public static final String relative_url = "relative_url";
    public static final String icon = "icon";
    public static final String promo_amount = "promo_amount";
    public static final String promo_code = "promo_code";

    /*NumberVerifyRequest*/
    public static final String mobile = "mobile";
    public static final String booking_amount = "booking_amount";
    public static final String country_code = "country_code";

    /*NumberVerifyResponse*/
    public static final String otp = "otp";
    public static final String expire_in = "expire_in";

    /*Social login Request*/
    public static final String type = "type";
    public static final String token = "token";
    public static final String device_type = "device_type";
    public static final String device_token = "device_token";
    public static final String referral_code = "referral_code";

    /*Social login Response*/
    public static final String first_name = "first_name";
    public static final String middle_name = "middle_name";
    public static final String last_name = "last_name";
    public static final String profile_url = "profile_url";
    public static final String referral_parameter = "referal_parameter";
    public static final String teke_teke_balance = "teke_teke_balance";

    /*Add Location Request */
    public static final String bookmark_type = "bookmark_type";
    public static final String location_name = "location_name";

    /* Add Location Response*/
    public static final String added_id = "added_id";
    public static final String callID = "calId";

    /* Get Route list Request */
    public static final String location = "location";
    public static final String lat = "lat";
    public static final String from = "from";
    public static final String to = "to";

    /* Get Route list Response */
    public static final String route_list = "routelist";
    public static final String route_name = "route_name";
    public static final String distance = "distance";
    public static final String stations = "stations";

    /* Station Details */
    public static final String station_id = "station_id";
    public static final String name = "name";
    public static final String seq_num = "seq_num";
    public static final String google_predicted_time = "google_predicted_time";
    public static final String estimate_time = "estimate_time";

    /* Ride Timing Request */
    public static final String route_id = "route_id";
    public static final String source_st_id = "source_station_id";
    public static final String destination_st_id = "destination_station_id";
    public static final String day_index = "day_index";
    public static final String current_date = "current_date";
    public static final String current_time = "current_time";

    /* Ride Timing Response */
    public static final String vehicle_list = "rides";

    /* Vehicle Details Model */
    public static final String status = "status";
    public static final String ride_price = "ride_price";
    public static final String amount_to_be_paid = "amount_to_be_paid";
    public static final String ride_id = "ride_id";
    public static final String remaining_amount = "remaining_amount";
    public static final String booking_fee = "booking_fee";
    public static final String full_amount = "full_amount";
    public static final String transaction_amount = "transaction_amount";

    /* Delete Bookmark Location Request */
    public static final String bookmark_location_id = "bookmark_location_id";

    /* Confirm Booking Request */
    public static final String is_lipamathree = "is_lipamathree";
    public static final String payment_method = "payment_method";
    public static final String date = "date";
    public static final String booking_date = "booking_date";
    public static final String time = "time";
    public static final String rider_name = "rider_name";
    public static final String no_of_booked_seat = "no_of_booked_seat";
    public static final String check_out_status = "check_out_status";

    /* Confirm Booking response */
    public static final String booking_id = "booking_id";
    public static final String booking_status = "booking_status";
    public static final String payment_status = "payment_status";
    public static final String ride_status = "ride_status";
    public static final String driver_info = "driverinfo";
    public static final String pickup_station_name = "pickup_station_name";
    public static final String stationes = "stations";
    public static final String ride_start_time = "ride_start_time";
    public static final String ride_end_time = "ride_end_time";
    public static final String pickUpObj = "pickUpObj";
    public static final String dropOffObj = "dropOffObj";
    public static final String pay_type = "pay_type";//pay_during_ride / pay_now
    public static final String pay_now_discount_per = "pay_now_discount_per";

    /* Booking History response */
    public static final String bookingdetails = "bookingdetails";
    public static final String upcoming = "upcoming";
    public static final String past = "past";
    public static final String vehicle_name = "vehicle_name";
    public static final String vehicle_type = "vehicle_type";
    public static final String vehicle_code = "vehicle_code";
    public static final String vehicle_model = "vehicle_model";
    public static final String driver_id = "driver_id";
    public static final String vehicle_img = "vehicle_img";
    public static final String driverinfo = "driverinfo";
    public static final String arrived_time = "arrived_time";
    public static final String boarding_pass = "boarding_pass";
    public static final String vehicle_capacity = "vehicle_capacity";
    public static final String booking_count = "booking_count";
    public static final String ride_date = "ride_date";
    public static final String user_id = "user_id";
    public static final String vehicle_reg_no = "vehicle_reg_no";

    /* Cancel Ride Request */
    public static final String rider_id = "rider_id";
    public static final String pickupStationId = "pickupStationId";
    public static final String dropOffStationId = "dropOffStationId";
    public static final String cancel_reason = "cancel_reason";
    public static final String cancel_date = "cancel_date";
    public static final String cancel_time = "cancel_time";
    public static final String rating = "rating";
    public static final String from_lat = "from_lat";
    public static final String from_location = "from_location";
    public static final String from_long = "from_long";
    public static final String to_lat = "to_lat";
    public static final String to_long = "to_long";
    public static final String to_location = "to_location";

    /* Trip Issue Request */
    public static final String booking_time = "booking_time";
    public static final String dropoff_station_name = "dropoff_station_name";
    public static final String complaint_type = "complaint_type";
    public static final String complaint = "complaint";

    /* Referral List Response */
    public static final String total_earning = "total_earning";
    public static final String total_invites = "total_invites";
    public static final String users = "users";
    public static final String email = "email";
    public static final String profile_img = "profile_img";
    public static final String earning_amount = "earning_amount";
    public static final String created_at = "created_at";

    /* Promo Code Response */
    public static final String promo_value = "promo_value";

    /* Transactions History Request */
    public static final String from_date = "from_date";
    public static final String to_date = "to_date";

    /* Transaction History Response */
    public static final String txn_type = "txn_type";
    public static final String amount = "amount";
    public static final String transaction_date = "transaction_date";
    public static final String description = "description";
    public static final String txn_date_time = "txn_date_time";
    public static final String transaction_history = "txn_history";
    public static final String notification_id = "notification_id";
    public static final String title = "title";
    public static final String msg = "msg";
    public static final String date_time = "date_time";
    public static final String notifications = "notifications";
    public static final String pay_number = "pay_number";

    /*........................MPESA payment request....................*/
    public static final String device_id = "device_id";
    public static final String payer_mobile_number = "payer_mobile_number";
    public static final String account_number = "account_number";
    public static final String currency = "currency";
    public static final String payment_request_reference_UUID = "payment_request_reference_UUID";
    public static final String payment_origin = "payment_origin";
    public static final String payment_type = "payment_type";
    public static final String payment_source = "payment_source";
    public static final String reference_id = "reference_id";
}

File: app/src/main/java/com/baseapplication/util/SerializedNameKeys.kt
package com.baseapplication.util

object SerializedNameKeys {
    val key: String = "key"
    val statusCode: String = "code"
    val statusMessage: String = "message"
    const val data: String = "data"
    const  val id: String = "id"
    const  val riderId: String = "rider_id"
    val latitude: String = "latitude"
    val longitude: String = "longitude"
    val address: String = "address"
    val created: String = "created_at"
    val updated: String = "updated_at"
    val locations: String = "locations"
    val relative_url: String = "relative_url"
    val icon: String = "icon"
    val promo_amount: String = "promo_amount"
    val promo_code: String = "promo_code"

    /*NumberVerifyRequest*/
    const val mobile: String = "mobile"
    val booking_amount: String = "booking_amount"
    const val country_code: String = "country_code"

    /*NumberVerifyResponse*/
    val otp: String = "otp"
    val expire_in: String = "expire_in"

    /*Social login Request*/
    val type: String = "type"
    const val token: String = "token"
    val device_type: String = "device_type"
    val device_token: String = "device_token"
    val referral_code: String = "referral_code"

    /*Social login Response*/
    const val first_name: String = "first_name"
    const val middle_name: String = "middle_name"
    const val last_name: String = "last_name"
    const    val profile_url: String = "profile_url"
    const val referral_parameter: String = "referal_parameter"
    const  val teke_teke_balance: String = "teke_teke_balance"

    /*Add Location Request */
    val bookmark_type: String = "bookmark_type"
    val location_name: String = "location_name"

    /* Add Location Response*/
    val added_id: String = "added_id"
    val callID: String = "calId"

    /* Get Route list Request */
    const  val location: String = "location"
    val lat: String = "lat"
    const val from: String = "from"
    const val to: String = "to"

    /* Get Route list Response */
    val route_list: String = "routelist"
    val route_name: String = "route_name"
    val distance: String = "distance"
    val stations: String = "stations"

    /* Station Details */
    val station_id: String = "station_id"
    const val name: String = "name"
    val seq_num: String = "seq_num"
    val google_predicted_time: String = "google_predicted_time"
    val estimate_time: String = "estimate_time"

    /* Ride Timing Request */
    val route_id: String = "route_id"
    val source_st_id: String = "source_station_id"
    val destination_st_id: String = "destination_station_id"
    val day_index: String = "day_index"
    val current_date: String = "current_date"
    val current_time: String = "current_time"

    /* Ride Timing Response */
    val vehicle_list: String = "rides"

    /* Vehicle Details Model */
    val status: String = "status"
    val ride_price: String = "ride_price"
    val amount_to_be_paid: String = "amount_to_be_paid"
    val ride_id: String = "ride_id"
    val remaining_amount: String = "remaining_amount"
    val booking_fee: String = "booking_fee"
    val full_amount: String = "full_amount"
    val transaction_amount: String = "transaction_amount"

    /* Delete Bookmark Location Request */
    val bookmark_location_id: String = "bookmark_location_id"

    /* Confirm Booking Request */
    val is_lipamathree: String = "is_lipamathree"
    val payment_method: String = "payment_method"
    val date: String = "date"
    val booking_date: String = "booking_date"
    val time: String = "time"
    val rider_name: String = "rider_name"
    val no_of_booked_seat: String = "no_of_booked_seat"
    val check_out_status: String = "check_out_status"

    /* Confirm Booking response */
    val booking_id: String = "booking_id"
    val booking_status: String = "booking_status"
    val payment_status: String = "payment_status"
    val ride_status: String = "ride_status"
    val driver_info: String = "driverinfo"
    val pickup_station_name: String = "pickup_station_name"
    val stationes: String = "stations"
    val ride_start_time: String = "ride_start_time"
    val ride_end_time: String = "ride_end_time"
    val pickUpObj: String = "pickUpObj"
    val dropOffObj: String = "dropOffObj"
    val pay_type: String = "pay_type" //pay_during_ride / pay_now
    val pay_now_discount_per: String = "pay_now_discount_per"

    /* Booking History response */
    val bookingdetails: String = "bookingdetails"
    val upcoming: String = "upcoming"
    val past: String = "past"
    val vehicle_name: String = "vehicle_name"
    val vehicle_type: String = "vehicle_type"
    val vehicle_code: String = "vehicle_code"
    val vehicle_model: String = "vehicle_model"
    val driver_id: String = "driver_id"
    val vehicle_img: String = "vehicle_img"
    val driverinfo: String = "driverinfo"
    val arrived_time: String = "arrived_time"
    val boarding_pass: String = "boarding_pass"
    val vehicle_capacity: String = "vehicle_capacity"
    val booking_count: String = "booking_count"
    val ride_date: String = "ride_date"
    val user_id: String = "user_id"
    val vehicle_reg_no: String = "vehicle_reg_no"

    /* Cancel Ride Request */
    val rider_id: String = "rider_id"
    val pickupStationId: String = "pickupStationId"
    val dropOffStationId: String = "dropOffStationId"
    val cancel_reason: String = "cancel_reason"
    val cancel_date: String = "cancel_date"
    val cancel_time: String = "cancel_time"
    val rating: String = "rating"
    val from_lat: String = "from_lat"
    val from_location: String = "from_location"
    val from_long: String = "from_long"
    val to_lat: String = "to_lat"
    val to_long: String = "to_long"
    val to_location: String = "to_location"

    /* Trip Issue Request */
    val booking_time: String = "booking_time"
    val dropoff_station_name: String = "dropoff_station_name"
    val complaint_type: String = "complaint_type"
    val complaint: String = "complaint"

    /* Referral List Response */
    val total_earning: String = "total_earning"
    val total_invites: String = "total_invites"
    val users: String = "users"
    const val email: String = "email"
    val profile_img: String = "profile_img"
    val earning_amount: String = "earning_amount"
    val created_at: String = "created_at"

    /* Promo Code Response */
    val promo_value: String = "promo_value"

    /* Transactions History Request */
    val from_date: String = "from_date"
    val to_date: String = "to_date"

    /* Transaction History Response */
    val txn_type: String = "txn_type"
    val amount: String = "amount"
    val transaction_date: String = "transaction_date"
    val description: String = "description"
    val txn_date_time: String = "txn_date_time"
    val transaction_history: String = "txn_history"
    val notification_id: String = "notification_id"
    val title: String = "title"
    val msg: String = "msg"
    val date_time: String = "date_time"
    val notifications: String = "notifications"
    val pay_number: String = "pay_number"

    /*........................MPESA payment request....................*/
    val device_id: String = "device_id"
    val payer_mobile_number: String = "payer_mobile_number"
    val account_number: String = "account_number"
    val currency: String = "currency"
    val payment_request_reference_UUID: String = "payment_request_reference_UUID"
    val payment_origin: String = "payment_origin"
    val payment_type: String = "payment_type"
    val payment_source: String = "payment_source"
    val reference_id: String = "reference_id"
}
